blocks/library/paragraph/index.js

```diff
@@ -25,8 +25,186 @@ import ToggleControl from '../../inspector-controls/toggle-control';
 import RangeControl from '../../inspector-controls/range-control';
 import ColorPalette from '../../color-palette';
 import BlockDescription from '../../block-description';
+import ContrastChecker from '../../contrast-checker';
 
 const { children } = source;
+const { getComputedStyle } = window;
+
+class ParagraphBlock extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.editableRef = null;
+
+		this.state = {
+			fallbackTextColor: undefined,
+		};
+
+		this.bindRef = this.bindRef.bind( this );
+		this.toggleDropCap = this.toggleDropCap.bind( this );
+	}
+
+	toggleDropCap() {
+		const { attributes, setAttributes } = this.props;
+		setAttributes( { dropCap: ! attributes.dropCap } );
+	}
+
+	bindRef( node ) {
+		if ( ! node ) {
+			return;
+		}
+		this.editableRef = node.querySelector( '[contenteditable="true"]' );
+	}
+
+	componentDidMount() {
+		this.grabTextColor();
+	}
+
+	componentDidUpdate() {
+		this.grabTextColor();
+	}
+
+	grabTextColor() {
```

Comment:
@jorgefilipecosta have you considered factoring out color related logic into a higher-order component? It looks very familiar to the implementation of the button. In both cases, we have text color and background color that are compared. From what I see this wrapper component could handle:
- node
- editableRef
- state
- bindRef
- grabTextColor
- componentDidMount & componentDidUpdate